File: Homework1/Problem1/Matlab/Problem1.m
efile=fopen('errorvaluesmatlab.txt','w');
hfile=fopen('hvaluesmatlab.txt','w');

t=2; %time

for iter=0:20 %defines y values
    h=10^(-iter);
    e=abs(derivative_of_function(t) - (1/h).*(func(t+h)-func(t)));
    
    if iter==0
        fprintf(efile,'%.20f',e)
        fprintf(hfile,'%.20f',h)
    end
    
    if iter>1
        fprintf(efile,',%.20f',e)
        fprintf(hfile,',%.20f',h)
    end
    
end


edata=load('errorvaluesmatlab.txt');
hdata=load('hvaluesmatlab.txt');

f=figure

set(f,'Visible','on')

loglog(hdata,edata)

function k=func(t)
    k=1./(1+9*exp(-t));
    %defines the function
end
function d=derivative_of_function(t)
    d=9.*exp(-t).*func(t)^2;
    %defines the functions derivative
end

File: Homework1/Problem1/Python/Problem1.py
import numpy as np
import pandas as pd
import matplotlib as mp
import matplotlib.pyplot as plt

pfile=open('errorspython.txt','w') #opens a txt file for error values
hfile=open('hvaluespython.txt','w') #opens a txt file for h values

def function(t):
    return 1/(1+9*np.exp(-t))
    #defines the function

def derivative_of_function(t):
    return 9*np.exp(-t)*function(t)**2
    #defines the functions derivative

t=2 #time

for iter in range(21): #defines y values
    h=10**(-iter)
    e=abs(derivative_of_function(t) - (1/h)*(function(t+h)-function(t)))

    if iter==0:
        pfile.write(str(e))

    if iter>0:
        pfile.write(","+str(e))

    
for niter in range(21): #defines x values
    m=10**(-niter)

    if niter==0:
        hfile.write(str(m))

    if niter>0:
        hfile.write(","+str(m))

pfile.close()
hfile.close()

edata=np.genfromtxt('errorspython.txt',delimiter=',')
hdata=np.genfromtxt('hvaluespython.txt',delimiter=',')

plt.scatter(hdata,edata)
plt.plot(hdata,edata)

ax=plt.gca()
ax.set_xscale('log')
ax.set_yscale('log')
ax.set_xlabel('delta t')
ax.set_ylabel('absolute error')
ax.set_title('Absolute Error Plot')

plt.show();
#show the plot
plt.savefig('pythonplot.eps', format='eps')
#save the figure.



plt.cla(); 
#this clears all data in the previous plot. 





